### PR TITLE
folder level validation

### PR DESCRIPTION
**Pull request by ruowan**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -532,25 +552,51 @@ const avocadoForDir = async (cwd: string) => {
...
-    await pr.checkout(pr.sourceBranch)
-    const sourceMap = await avocadoForDir(pr.workingDir)
+    for (const dir of readmeDirs) {
+      await pr.checkout(pr.targetBranch)
+      const targetMap = await avocadoForDir(path.resolve(pr.workingDir, dir))
 
-    const fileChanges = await pr.diff()
+      // collect all errors from the 'sourceBranch'
+      await pr.checkout(pr.sourceBranch)
+      const sourceMap = await avocadoForDir(path.resolve(pr.workingDir, dir))
```
Comment (jianyexi) [inline src/index.ts:585]: What if the source branch does not have the dir ?

Comment (ruowan) [reply src/index.ts:585]: it will return empty map and only report new errors.

**Review by jianyexi — APPROVED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -532,25 +552,51 @@ const avocadoForDir = async (cwd: string) => {
...
 
-    // remove existing errors.
-    /* Note: For MISSING_README error if the error is related to the PR changes,
+      const fileChanges = await pr.diff()
+
+      // remove existing errors.
+      /* Note: For MISSING_README error if the error is related to the PR changes,
```
Comment (raych1) [inline src/index.ts:590]:
```suggestion
```

**Review by raych1 — APPROVED** [review]

**Pull request merged**